### PR TITLE
Fix issue when transforming DTOs with nested DTOs into array or json

### PR DESCRIPTION
**Pull request by WendellAdriel**
This PR fixes the issue reported on #50

### CODE REVIEW

**Comment by Briareos17** [comment]
Big thanks for fast response, but what about collection of DTOs? toArray simply return array ot DTO instances but each need to be array

**Comment by Briareos17** [comment]
Would be great to implement some kind of reverse casting, so all typehinted or casted properties will cast back on serialize

**Comment by WendellAdriel** [comment]
Oooh, completely forgot about the Collections 😅 
I'll add a fix for that as well, now the reverse casting is not something I plan to add right now to the package

File: src/SimpleDTO.php [hunk 1]
```diff
@@ -399,15 +398,26 @@ private function isArrayable(mixed $value): bool
...
     {
-        if (is_array($value)) {
-            return $value;
-        }
+        return match (true) {
+            is_array($value) => $value,
+            $value instanceof Collection => $value->toArray(),
+            $value instanceof Model => $value->toArray(),
+            $value instanceof SimpleDTO => $this->transformDTOToArray($value),
+            is_object($value) => (array) $value,
```
Comment (simon-tma) [inline src/SimpleDTO.php:406]: This has broken Enums. A UnitEnum is now `['name' => 'Foo']` and a BackedEnum is now `['name' => Foo, 'value' => 'Bar']`

**Pull request merged**